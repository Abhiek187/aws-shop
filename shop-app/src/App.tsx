import { CssBaseline, ThemeProvider, createTheme } from "@mui/material";
import { teal, yellow } from "@mui/material/colors";
import { Route, Routes, useLocation } from "react-router-dom";

import TopBar from "./components/app-bar/TopBar";
import Store from "./components/store/Store";
import { selectApp } from "./store/appSlice";
import Profile from "./components/profile/Profile";
import NotFound from "./components/404/NotFound";
import { useAppSelector } from "./store/hooks";

const App = () => {
  const { mode } = useAppSelector(selectApp);
  const location = useLocation();

  const theme = createTheme({
    palette: {
      mode,
      primary: {
        main: teal[500],
        light: "#33ab9f",
        dark: "#00695f",
      },
      secondary: {
        main: yellow[700],
        light: "#ffde33",
        dark: "#b29500",
      },
      contrastThreshold: 4.5, // minimum contrast for a11y
    },
  });

  return (
    <ThemeProvider theme={theme}>
      {/* Have the background color match the theme mode */}
      <CssBaseline />
      <Routes location={location}>
        <Route
          path="/"
          element={
            <>
              <header className="sticky top-0 z-10">
                <TopBar />
              </header>
              <main className="m-2">
                <Store />
              </main>
            </>
          }
        />
        <Route path="/profile" element={<Profile />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </ThemeProvider>
  );
};

export default App;
