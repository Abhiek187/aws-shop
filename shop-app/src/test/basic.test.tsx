import React from "react";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import App from "../App";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test("Renders Vite default page", () => {
  const component = renderer.create(<App />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();

  const testInstance = component.root;
  const countButton = testInstance.findByProps({ className: "read-the-docs" });
  expect(countButton).toBeTruthy();
});
