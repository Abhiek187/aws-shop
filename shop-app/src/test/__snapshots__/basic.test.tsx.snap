// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders Vite default page 1`] = `
[
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        className="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        className="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>,
  <h1>
    Vite + React
  </h1>,
  <div
    className="card"
  >
    <button
      onClick={[Function]}
    >
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>,
  <p
    className="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>,
]
`;
